runtime/vm/visible.c

```diff
@@ -100,7 +106,8 @@ checkVisibility(J9VMThread *currentThread, J9Class* sourceClass, J9Class* destCl
 	if (!J9CLASS_IS_EXEMPT_FROM_VALIDATION(sourceClass)) {
 		if ( modifiers & J9AccPublic ) {
 			/* Public */
-			if ((sourceClass != destClass)
+			if (doModuleAccessCheck
```

Comment:
The flag can be used here

Reply:
yeah, check via `J9_ARE_NO_BITS_SET(lookupOptions, J9_LOOK_NO_MODULE_CHECKS)`.